src/daemon/gateway.go

```diff
@@ -18,17 +19,22 @@ import (
 	"github.com/skycoin/skycoin/src/visor/historydb"
 )
 
+const (
+	// AllAPISets special API set containing all exported methods
+	AllAPISets = "ALL"
+)
+
 // GatewayConfig configuration set of gateway.
 type GatewayConfig struct {
-	BufferSize      int
-	EnableWalletAPI bool
+	BufferSize     int
+	EnabledAPISets collections.StringSet
```

Comment:
Do not change anything in gateway. Handle it all in `api/`.  In skycoin/skycoin.go or skycoin/config.go, during config processing, you can set the gateway.EnableWalletAPI flag from the API set data

Reply:
This is used by `Gateway` instances to query for wallet API enabled e.g. [for wallet balance](https://github.com/skycoin/skycoin/blob/develop/src/daemon/gateway.go#L952-L954). Not sure it can be completely removed since methods are (re)used at multiple locations.